src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -778,701 +231,68 @@ public JdbcConnection getCurrentConnection() {
-      this.explicitlyAutoCommit = (Boolean) args[0];
-      this.inTransaction = !this.explicitlyAutoCommit;
-    }
-
-    if (METHOD_COMMIT.equals(methodName) || METHOD_ROLLBACK.equals(methodName)) {
-      this.inTransaction = false;
-    }
-
-    if (METHOD_SET_READ_ONLY.equals(methodName)) {
-      this.explicitlyReadOnly = (Boolean) args[0];
-      this.log.logTrace(
-          Messages.getString(
-              "ClusterAwareConnectionProxy.10",
-              new Object[] {"explicitlyReadOnly", this.explicitlyReadOnly}));
-      connectToWriterIfRequired(this.explicitlyReadOnly);
-    }
-  }
-
-  private void connectToWriterIfRequired(Boolean readOnly) throws SQLException {
-    if (shouldReconnectToWriter(readOnly) && !Util.isNullOrEmpty(this.hosts)) {
-      try {
-        connectTo(WRITER_CONNECTION_INDEX);
-      } catch (SQLException e) {
-        failover(WRITER_CONNECTION_INDEX);
       }
+      throw new IllegalStateException(e.getMessage(), e);
     }
   }
 
-  private boolean shouldReconnectToWriter(Boolean readOnly) {
-    return readOnly != null && !readOnly && !isWriterHostIndex(this.currentHostIndex);
-  }
-
-  /**
-   * Closes current connection.
-   *
-   * @throws SQLException if an error occurs
-   */
-  @Override
-  protected synchronized void doClose() throws SQLException {
-    try {
-      this.pluginManager.execute(this.getClass(), "close", () -> {
-        this.currentConnection.close();
-        return null;
-      });
-    } catch (SQLException sqlEx) {
-      throw sqlEx;
-    } catch (Exception ex) {
-      throw new SQLException(ex.getMessage(), ex);
-    }
-    releasePluginManager();
-  }
-
-  /**
-   * Aborts current connection using the given executor.
-   *
-   * @throws SQLException if an error occurs
-   */
-  @Override
-  protected synchronized void doAbort(Executor executor) throws SQLException {
-    try {
-      this.pluginManager.execute(this.getClass(), "abort", () -> {
-        this.currentConnection.abort(executor);
-        return null;
-      });
-    } catch (SQLException sqlEx) {
-      throw sqlEx;
-    } catch (Exception ex) {
-      throw new SQLException(ex.getMessage(), ex);
-    }
-    releasePluginManager();
-  }
-
-  /**
-   * Aborts current connection.
-   *
-   * @throws SQLException if an error occurs
-   */
-  @Override
-  protected synchronized void doAbortInternal() throws SQLException {
-    try {
-      this.pluginManager.execute(this.getClass(), "abortInternal", () -> {
-        this.currentConnection.abortInternal();
-        return null;
-      });
-    } catch (SQLException sqlEx) {
-      throw sqlEx;
-    } catch (Exception ex) {
-      throw new SQLException(ex.getMessage(), ex);
-    }
-    releasePluginManager();
-  }
-
-  @Override
-  protected InvocationHandler getNewJdbcInterfaceProxy(Object toProxy) {
-    return new JdbcInterfaceProxy(toProxy);
-  }
+  protected synchronized Object invokeMore(Object proxy, Method method, Object[] args)
```

Comment:
I think invokeMore() is getting smaller and smaller and could be merged into invoke().